backend-es/test/snapshots-out/Snapshot.FunctionCompose03.js

```diff
@@ -14,10 +14,10 @@ const test3 = f => g => {
   return x => $0($1($2($3(x))));
 };
 const test2 = f => g => {
-  const $0 = g();
-  const $1 = f();
+  const $0 = f();
+  const $1 = g();
   const $2 = g();
-  return x => $0($1($2(x)));
+  return x => $2($0($1(x)));
```

Comment:
The ordering here seems strange to me. I believe the original code was ordered in an obvious way per the original syntax. I understand that evaluation order isn't strictly defined, but I do have concerns that too much movement makes the output diverge considerably from the source in such a way to inhibit reasoning in tricky situations. The concern I have about something like this sort of movement is I'm not sure how this will translate to larger programs.